### PR TITLE
[legend][scale] add @vx/legend, add additional scales fixes #65

### LINKED ISSUE #65: [legend] add @vx/legend

### PR DESCRIPTION
**Pull request by hshoff**
### Adds:

- **`@vx/scale`**
  + `scaleQuantize()`
  + `scaleQuantile()`
  + `scaleThreshold()`
- **`@vx/legend`**
  + `<LegendLinear />`
  + `<LegendOrdinal />`
  + `<LegendQuantile />`
  + `<LegendThreshold />`

screen shot 2017-06-14 at 9 34 40 pm

[view source](https://github.com/hshoff/vx/blob/568b705fc7b8a0f8596f43dbd61398cc2b085c65/packages/vx-demo/pages/legends.js)

`@vx/legend` uses HTML instead of SVG. Benefits of HTML for legends includes using flexbox for layout and easy to add controls like checkboxes, sliders, selects, etc.

Need to add:

  + `<LegendSize />` 
  + `shape` prop should be one of `rect | circle | Element | function`
  + pass additionalProps to LabelItem for click handlers, pass through props

fixes #65

### CODE REVIEW

File: packages/vx-legend/src/legends/Legend.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  itemMargin = '0',
+  direction = 'column',
+  itemDirection = 'row',
+  ...restProps,
+}) {
+  return (
+    <div
+      className={cx('vx-legend', className)}
+      style={style}
+      style={{
```
Comment (williaster) [inline packages/vx-legend/src/legends/Legend.js]: not sure if this is intentional but looks like the passed `style` would be overridden here? I think passing styles would be solid though, then you could `flex-wrap` vs not, etc.

Comment (hshoff) [inline packages/vx-legend/src/legends/Legend.js]: not intentional, i've got this fixed locally, just haven't pushed it yet. Good catch! 👍

**Comment by hshoff** [comment]
Update:

`<LegendSize />`

screen shot 2017-06-20 at 8 13 29 am

custom shapes

screen shot 2017-06-20 at 8 13 11 am

Almost there.

**Comment by hshoff** [comment]
closing in favor of https://github.com/hshoff/vx/pull/77

**Pull request closed without merging**